flink-state-backends/flink-statebackend-forst/src/main/java/org/apache/flink/state/forst/ForStKeyedStateBackendBuilder.java

```diff
@@ -298,6 +298,9 @@ public ForStKeyedStateBackend<K> build() throws BackendBuildingException {
             IOUtils.closeQuietly(restoreOperation);
             try {
                 optionsContainer.clearDirectories();
+                // TODO: Remove this after FLINK-37442, if we could properly handl the directory
+                //       deletion in file mapping manager.
+                optionsContainer.forceClearRemoteDirectories();
```

Comment:
`clearDirectories` has removed `remoteBasePath`?
Is `forceClearRemoteDirectories` necessary here?

Reply:
No, the `clearDirectories` has not removed `remoteBasePath`, after we add [this](https://github.com/apache/flink/pull/26279/files#diff-e5f6069abc8b000724d66d30f841fc51c0c2746726a58ca6b62f5c8e86d109efR282). All the directory deletion will be disabled after this (since it is `NOT_OWNED`), and if the builder build failed, we treat the `remoteBasePath` specially.